centinel/primitives/dnslib.py

```diff
@@ -94,12 +94,8 @@ def lookup_domains(self):
         """
         for nameserver in self.nameservers:
             for domain in self.domains:
-                escape_count = 0
                 while threading.active_count() > self.max_threads:
                     time.sleep(1)
-                    escape_count += 1
-                    if escape_count > 100:
```

Comment:
I have found the issue and fixed it. It was a bug in the way we join threads: execution moves on after the last thread has joined (it will not wait on other running threads) and the results arrive incomplete.